Issue 3 - Fix unit test issues with PREFIX, imports, and version

droideck opened the pull request:
Description: Currently, PREFIX is set to '/' by default when PREFIX
env var is not set. It leads to an error when we point to a wrong dir.
Also, fix test imports and make it use the current Python version
instead of 3.9.

droideck commented:
The playbook doesn't work on my current setup currently, giving me the next error:

```
>       return self.result['my_useful_info']['instances'][instname][attr]
```

Probably, I've messed up my env but I need to figure out how to make our code more error-prone and document a bit more.

progier389 commented on ansible_collections/ds/ansible_ds/tests/conftest.py:
```diff
@@ -71,7 +71,7 @@ def isPrefix(dir):
             return False
 
-    def setPrefix(dir, lib389dir):
+    def setPrefix(dir):
```
IMHO we should have a test here with 2 paths:
if dir == '/':
   os.environ.pop('PREFIX', None)
   lp= '/usr' + lp
else:
   os.environ['PREFIX'] = dir

progier389 commented on Makefile:
```diff
@@ -30,6 +30,6 @@ install:
 	cd ansible_collections ; ansible-galaxy collection install $B -f
 
 unit_test: clean $B install
-	cd ansible_collections/$N/$M ; ansible-test units -vvvvv --python 3.9 --local
```
Why ?
FYI:  removing it break things on my machine:
  /usr/bin/python2.7: No module named pytest

droideck replied:
I think we really shouldn't hard code the Python version here because then we lock our tests to be run with the exact version (3.10 won't work, etc.).

So something is fishy with your Python environment, I think, as it should discover the latest python version automatically. Feel free to ping me so we can work out the best way to fix it. (may be adding some additional processing here for prefix installs...)

droideck commented on ansible_collections/ds/ansible_ds/tests/conftest.py:
```diff
@@ -65,52 +64,47 @@ def ansibletest(caplog):
-            pass
-        if ignore != None:
-            lookup.append( ( str(Path(dir).parent), str(Path(dir).name)) )
+    #lookup = [ ( Path(__file__).parent.parent.parent.parent.parent, 'ansible-ds'), ]
```
Please, see my next answer.

droideck commented:
I've also moved roles because by default it looks for them in the directory with playbooks:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-directory-structure

droideck commented:
Okay, I think it's good enough for the final review.
I've added Readme and I fixed Python version while running tests.

ansible-test runs all of the available Python versions so I've forced the default Python version, which should be Python 3 (we don't support Python 2).

progier389 commented on Makefile:
```diff
@@ -30,6 +30,6 @@ install:
 
 unit_test: clean $B install
-	cd ansible_collections/$N/$M ; ansible-test units -vvvvv --local
+	cd ansible_collections/$N/$M ; ansible-test units -vvvvv --python ${python_version}--local
```
missing space before --local ?

progier389 reviewed (changes requested):
Still to mandatory changes ( in Makefile and test_ds_info )
should also discuss if we should  really move the role directory within playbooks

droideck commented:
> Note sure we should move the roles inside playbooks directory as the trend is to put it in the collection root (cf https://www.ansible.com/blog/the-future-of-ansible-content-delivery?extIdCarryOver=true&sc_cid=701f2000001Css5AAC )

Thanks for the article; I haven't seen it!

The issue is that if I don't move the `roles`, it fails to find the required functions when I run the playbook.
It is on a clean install on 1minutetip.
So we need to do some additional actions and configurations to make it work as in the article.

This PR is already covered too much, so I just made it as in the official docs:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-directory-structure 
Because by default, when you run the playbook, it looks for roles in the directory with playbooks.

We can do something like ansible-freeipa - add a section on how to configure `ansible.cfg`, etc.
Or something else. But I think it's a point for a separate ticket.
What do you think?

droideck commented on ansible_collections/ds/ansible_ds/tests/unit/plugins/modules/test_ds_info.py:
```diff
@@ -1,22 +1,28 @@
-# Set PYTHONPATH to be able to find lib389 (computed within conftest seesion initialization hook)
-sys.path.insert(0, os.environ['LIB389PATH'])
+# Set PYTHONPATH to be able to find lib389 in PREFIX install (computed within conftest seesion initialization hook)
+if "PREFIX" in os.environ:
```
Good catch!

progier389 commented:
OK let it has you have written it since that is what works in 1minutesTips environment 
   ( I suspect that the doc I pointed is fr future ansible version and we will adapt when it is out ... )
 we could see if we get in trouble with other ansible version (maybe some doc as you said or using full qualified names rather than short names )

progier389 reviewed (approved):
LGTM
